### PR TITLE
Add conditional orders (inverse and linear) WIP

### PR DESCRIPTION
**Pull request by peepopoggers**
Cargo clippy and fmt both are fine, however this needs testing and the relevant issues for conditional orders need solving.

### CODE REVIEW

**Comment by peepopoggers** [comment]
I have tested conditional orders, which are working fine, however something to be noted, the default trigger_by value (Unknown as default) parameter is not accepted so Some(TriggerPrice::LastPrice) needs to be passed instead.

Have not tested inverse orders yet but I have opened and processed linear conditional orders which seem to be working fine.

**Comment by wassup-** [comment]
This looks very good! Can you tell me what needs to happen still? I may be able to spend some time on this to speed up development.
I wouldn't worry too much about extensively testing inverse orders; if users of this library run into any issues they should report those issues so we can fix them.

**Comment by peepopoggers** [comment]
Linear orders are working fine and unless the response queries are incorrect for inverse, i think inverse orders are good too (if there are any problems it will be a 1 minute fix), so I think this is ready for review. You can test this out if you want, example below.

```rs
//"api" is just http rest client

let q_symbols = api.fetch_symbols().await?;
let symbol_info = q_symbols.symbols().into_iter().find(|symbol| symbol.alias == *pair).unwrap(); //pair is pair you want to search
let qty_step = symbol_info.lot_size_filter.qty_step; //this is minimum step for order size, e.g 0.001, 0.002 (0.0009 or 0.0015 not allowed)

let qty = ? //whatever you want

//change quantity to buy to stepped version so bybit api can accept it
qty_to_buy = (qty_to_buy/qty_step).round() * qty_step;

let order_data = PlaceLinearConditionalOrderData {
    symbol: pair.to_string(),
    side: Side::Buy,
    qty: qty_to_buy, //must follow qty step guidelines
    order_type: OrderType::Market,
    price: price_to_buy,
    time_in_force: TimeInForce::PostOnly,
    take_profit: Some(takeprofit),
    stop_loss: Some(stoploss),
    reduce_only: false,
    close_on_trigger: false,
    base_price: last_price, //most of the time just use last price (it is used to calculate direction of stop order)
    stop_px: entry,
    trigger_by: Some(TriggerPrice::LastPrice), //this is required even though it is an option type
    ..Default::default()
};

//debug
println!("{:#?}", order_data);
let order_result = api.place_linear_conditional_order(order_data).await?;
```

This is not test-ready due to context but this should make it easier to understand implementation. For inverse, place_conditional_order(PlaceActiveOrderData) is used.

**Comment by peepopoggers** [comment]
Hey, I think this should be merged as-is, it works well for me.

File: src/rest/orders.rs [hunk 1]
```diff
@@ -63,8 +63,36 @@ pub struct PlaceActiveOrderData {
...
+    pub order_type: OrderType,
+    pub price: Option<f64>,
+    pub time_in_force: TimeInForce,
+    pub close_on_trigger: bool,
+    pub order_link_id: Option<OrderLinkId>,
+    pub take_profit: Option<f64>,
+    pub stop_loss: Option<f64>,
+    pub tp_trigger_by: Option<TriggerPrice>,
+    pub sl_trigger_by: Option<TriggerPrice>,
+    //conditional specifics
```
Comment (wassup-) [inline src/rest/orders.rs:89]: could you insert a space between `//` and the actual comment?

File: src/rest/orders.rs [hunk 2]
```diff
@@ -63,8 +63,36 @@ pub struct PlaceActiveOrderData {
...
+    pub base_price: f64,
+    pub stop_px: f64,
+    pub trigger_by: Option<TriggerPrice>,
+    //linear conditional specifics
```
Comment (wassup-) [inline src/rest/orders.rs:93]: could you insert a space between `//` and the actual comment?

File: src/rest/orders.rs [hunk 3]
```diff
@@ -248,6 +319,13 @@ mod request {
         pub reduce_only: Option<bool>,
         #[serde(skip_serializing_if = "Option::is_none")]
         pub position_idx: Option<i64>,
+        //conditional specifics
```
Comment (wassup-) [inline src/rest/orders.rs:322]: could you insert a space between `//` and the actual comment?

File: src/rest/orders.rs [hunk 4]
```diff
@@ -248,6 +319,13 @@ mod request {
         pub reduce_only: Option<bool>,
```
Comment (wassup-) [inline src/rest/orders.rs:319]: maybe we should insert the `// linear conditional specifics` comment here too for consistency?

**Review by wassup- — COMMENTED** [review]
Looks good! If you could find the time to address my comments, we can merge this 🙂

**Comment by peepopoggers** [comment]
Think everything is good now :>

**Comment by wassup-** [comment]
Thank you!

**Pull request merged**